test/package/packageTest.nut.ts

```diff
@@ -141,44 +135,41 @@ describe('Integration tests for #salesforce/packaging library', function () {
       pkgCreateVersionRequestId = result.Id;
     });
 
-    it('runs force:package:version:create:report until it succeeds', async () => {
-      let duration = 0;
-      let retries = 10;
-      async function pollForPackageCompletion(remainingRetries): Promise<PackageVersionCreateResponse> {
-        const pollResultRaw = execCmd<[PackageVersionCreateResponse]>(
-          `force:package:version:create:report --packagecreaterequestid ${pkgCreateVersionRequestId} --json`
-        );
-        if (!(getString(pollResultRaw, 'jsonOutput.jsonError.name') === 'JsonParseError')) {
-          const pollResult = pollResultRaw.jsonOutput.result[0];
-          expect(pollResult).to.include.keys(VERSION_CREATE_RESPONSE_KEYS);
-          // it's done! or timed out.
-          if (!VERSION_CREATE_STATUSES_INPROGRESS.includes(pollResult.Status) || remainingRetries <= 0) {
-            return pollResult;
-          }
-        } else if (retries > 0) {
-          // eslint-disable-next-line no-console
-          console.log(`JSON Parse Error\n${JSON.stringify(pollResultRaw, undefined, 2)}`);
-          retries--;
-        } else {
-          throw new Error('JsonParseError');
-        }
-        // still going...
-        duration += WAIT_INTERVAL_MS;
-        return sleep(WAIT_INTERVAL_MS * 2, Duration.Unit.MILLISECONDS).then(() =>
-          pollForPackageCompletion(remainingRetries - 1)
-        );
+    it('get package version create report', async () => {
+      const pv = new PackageVersion({ project, connection: devHubOrg.getConnection() });
+      const result = await pv.getCreateVersionReport(pkgCreateVersionRequestId);
+      expect(result).to.include.keys(VERSION_CREATE_RESPONSE_KEYS);
+
+      if (result.Status === 'Error') {
+        throw new Error(`pv.getCreateVersionReport failed with status Error: ${result.Error.join(';')}`);
       }
-      const result = await pollForPackageCompletion(LIST_ORG_GET_WAIT_MAX_TRIES);
+    });
+
+    it('wait for package version create to finish', async () => {
+      const pv = new PackageVersion({ project, connection: devHubOrg.getConnection() });
+      // "enqueued", "in-progress", "success", "error" and "timed-out"
+      Lifecycle.getInstance().on('enqueued', async (results: PackageVersionCreateReportProgress) => {
+        expect(results.Status).to.equal(PackagingSObjects.Package2VersionStatus.queued);
+      });
+      Lifecycle.getInstance().on('in-progress', async (results: PackageVersionCreateReportProgress) => {
+        // eslint-disable-next-line no-console
+        console.log(`in-progress: ${JSON.stringify(results, undefined, 2)}`);
```

Comment:
leftover `console.log`?

Reply:
No - just logging the payload in case the assertion fails in the test.